pytorch_lightning/utilities/enums.py

```diff
@@ -37,6 +39,32 @@ def __hash__(self) -> int:
         return hash(self.value.lower())
 
 
+class _OnAccessEnumMeta(EnumMeta):
+    """Enum with a hook to run a function whenever a member is accessed.
+
+    Adapted from:
+    https://www.buzzphp.com/posts/how-do-i-detect-and-invoke-a-function-when-a-python-enum-member-is-accessed
+    """
+
+    def __getattribute__(cls, name: str) -> Any:
+        obj = super().__getattribute__(name)
+        if isinstance(obj, Enum) and obj._on_access:
+            obj._on_access()
+        return obj
+
+    def __getitem__(cls, name: str) -> Any:
+        member = super().__getitem__(name)
+        if member._on_access:
+            member._on_access()
+        return member
+
+    def __call__(cls, value: str, *args: Any, **kwargs: Any) -> Any:
+        obj = super().__call__(value, *args, **kwargs)
+        if isinstance(obj, Enum) and obj._on_access:
+            obj._on_access()
```

Comment:
Can we use `deprecate` instead ? Do you see other reasons why we would like a more general _on_access function ?

Reply:
Updated!